### PR TITLE
Alter project structure to make test utilities explicit and update pa…

### LINKED ISSUE #12: Resolve build warnings
**Context**
Best practices is to require no warnings when building.

**Details**
- 3rd party libraries need to be updated.
- There is an error in the way the one of the test projects works, it currently hosts tests and test utilities, and another test references it. Restructure it so the test utilities are explicit.
- Unused field/s.

### PR DESCRIPTION
**Pull request by rwx-jacquier-robert**
…ckages.

# ✅ Pull Request Checklist

Before submitting your pull request, please ensure the following are complete:

## 🔍 Self Review
- [x] Confirmed that the destination branch is correct.
- [x] I have reviewed my own code carefully.
- [x] I have removed any debugging code or print statements.
- [x] Code changes are scoped to the purpose of this PR (no unrelated changes).
- [x] Comments are clear, and the code is self-explanatory where possible.
- [x] Documentation (if applicable) is updated.

## 📄 Task
- [x] The task/issue title is clear and descriptive.
- [x] The description explains **why** (context) and **what** (details) clearly.
- [x] Related issues are referenced (e.g., "Fixes #123").

## ✅ Code Quality
- [x] There are no new warnings or errors in the build output.
- [x] Code follows the project’s coding standards and formatting rules.

## 🧪 Testing
- [x] All unit and integration tests pass locally.
- [x] New functionality is covered by tests, or is explicitly marked as not testable.
- [x] Edge cases and error handling have been considered.

## ✅ CI
- [x] CI checks are passing (e.g., build, tests, linting).

## 📄 PR Quality
- [x] The PR title is clear and descriptive.
- [x] The PR description explains **what** and **why** clearly.
- [x] Related issues are referenced (e.g., "Fixes #123").

## 👥 Collaboration
- [x] I have assigned reviewers and/or requested feedback.
- [x] I’ve addressed or responded to all prior review comments (if applicable).

### CODE REVIEW

File: QuantSA/QuantSA.Excel.Addin.Tests/QuantSA.Excel.Addin.Tests.csproj
Comment (rwx-oberholzer-marcelle) [inline QuantSA/QuantSA.Excel.Addin.Tests/QuantSA.Excel.Addin.Tests.csproj:1]: Revert

File: QuantSA/QuantSA.Excel.AddIn/app.config [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
       </dependentAssembly>
       <dependentAssembly>
-        <assemblyIdentity name="Accord.Math.Core" publicKeyToken="fa1a88e29555ccf7" culture="neutral"/>
-        <bindingRedirect oldVersion="0.0.0.0-3.8.0.0" newVersion="3.8.0.0"/>
+        <assemblyIdentity name="Accord.Math.Core" publicKeyToken="fa1a88e29555ccf7" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.8.0.0" newVersion="3.8.0.0" />
       </dependentAssembly>
     </assemblyBinding>
   </runtime>
-<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.8"/></startup></configuration>
```
Comment (rwx-oberholzer-marcelle) [inline QuantSA/QuantSA.Excel.AddIn/app.config:19]: Need this

File: QuantSA/QuantSA.Excel.AddIn/QuantSA.Excel.Addin.csproj
Comment (rwx-oberholzer-marcelle) [inline QuantSA/QuantSA.Excel.AddIn/QuantSA.Excel.Addin.csproj:1]: Revert

File: QuantSA/QuantSA.Core/RootFinding/MultiDimNewton.cs
Comment (rwx-oberholzer-marcelle) [inline QuantSA/QuantSA.Core/RootFinding/MultiDimNewton.cs:1]: Revert

File: QuantSA/QuantSA.Core/RootFinding/BroydenWrapper.cs
Comment (rwx-oberholzer-marcelle) [inline QuantSA/QuantSA.Core/RootFinding/BroydenWrapper.cs:1]: Revert

**Review by rwx-oberholzer-marcelle — CHANGES_REQUESTED** [review]

File: QuantSA/QuantSA.Excel.Addin.Tests/QuantSA.Excel.Addin.Tests.csproj
Comment (rwx-jacquier-robert) [reply QuantSA/QuantSA.Excel.Addin.Tests/QuantSA.Excel.Addin.Tests.csproj:1]: Done

File: QuantSA/QuantSA.Excel.AddIn/app.config (hunk 1 above)
Comment (rwx-jacquier-robert) [reply QuantSA/QuantSA.Excel.AddIn/app.config:19]: Done

File: QuantSA/QuantSA.Excel.AddIn/QuantSA.Excel.Addin.csproj
Comment (rwx-jacquier-robert) [reply QuantSA/QuantSA.Excel.AddIn/QuantSA.Excel.Addin.csproj:1]: Done

File: QuantSA/QuantSA.Core/RootFinding/MultiDimNewton.cs
Comment (rwx-jacquier-robert) [reply QuantSA/QuantSA.Core/RootFinding/MultiDimNewton.cs:1]: Done

File: QuantSA/QuantSA.Core/RootFinding/BroydenWrapper.cs
Comment (rwx-jacquier-robert) [reply QuantSA/QuantSA.Core/RootFinding/BroydenWrapper.cs:1]: Done

**Review by rwx-oberholzer-marcelle — APPROVED** [review]
